File: Dynamic_Programming/20_Oct_2019_permutation_coefficient/20_Oct_2019_permutation_coefficient_dynamic.py
# dynamic permutation coeff

def permutationCoeff(n,k):
    fact = [0 for x in range(n+1)]
    fact[0] = 1

    for i in range(1,n+1):
        fact[i] = fact[i-1] * i
    
    return int(fact[n]/fact[n-k])

#driver code

n = 10
k = 2

print("Permuattion coeff of {} and {} is {}".format(n,k,permutationCoeff(n,k)))

File: Dynamic_Programming/october/20_Oct_2019_permutation_coefficient/20_Oct_2019_permutation_coefficient_dynamic.cpp
#include<bits/stdc++.h>
using namespace std;
int permutationCoeff(int n, int k){
    int fact[n+1];

    //base
    fact[0] = 1;

    for(int i = 1; i <= n; i++){
        fact[i] = i * fact[i-1];
    }
    // P(n,k) = n! / (n - k)! 
    return int (fact[n]/fact[n-k]);
}

//driver 
int main(){
    int n = 10, k = 2;

    cout << "Permuattion coeff of " << n << " and " << k << " is " << permutationCoeff(n,k);

    return 0;
}